src/main/java/com/ericsson/ei/frontend/BackEndInformationController.java

```diff
@@ -16,35 +16,107 @@
+import com.google.gson.JsonParser;
+import com.google.gson.reflect.TypeToken;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
-import org.springframework.web.bind.annotation.CrossOrigin;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
-import org.springframework.web.bind.annotation.RestController;
 
 import javax.servlet.http.HttpServletRequest;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
 
-@RestController
-public interface BackEndInformationController {
+@Controller
+public class BackEndInformationController {
+
+    @Autowired
+    private BackEndInformation backEndInformation;
+
+    @Autowired
+    private BackEndInstancesUtils backEndInstancesUtils;
 
-    @CrossOrigin
     @RequestMapping(value = "/get-instances", method = RequestMethod.GET)
-    ResponseEntity<String> getInstances(Model model);
+    public ResponseEntity<String> getInstances(Model model) {
+        return new ResponseEntity<>(backEndInstancesUtils.getInstances().toString(), HttpStatus.OK);
+    }
 
-    @CrossOrigin
     @RequestMapping(value = "/switch-backend", method = RequestMethod.POST)
-    ResponseEntity<String> switchBackEndInstance(Model model, HttpServletRequest request);
+    public ResponseEntity<String> switchBackEndInstance(Model model, HttpServletRequest request) {
+        try {
+            String body = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
+            backEndInstancesUtils.setInstances(new JsonParser().parse(body).getAsJsonArray());
+            backEndInstancesUtils.writeIntoFile();
+            for (BackEndInformation backEndInformation : backEndInstancesUtils.getInformation()) {
+                if (backEndInformation.isActive()) {
+                    backEndInstancesUtils.setBackEndProperties(backEndInformation);
+                }
+            }
+            return new ResponseEntity<>(HttpStatus.OK);
+        } catch (Exception e) {
+            return new ResponseEntity<>("Internal error", HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
 
-    @CrossOrigin
-    @RequestMapping(value = "/switch-backendByMainPage", method = RequestMethod.POST)
-    ResponseEntity<String> switchBackEndInstanceByMainPage(Model model, HttpServletRequest request);
-
-    @CrossOrigin
     @RequestMapping(value = "/switch-backend", method = RequestMethod.DELETE)
-    ResponseEntity<String> deleteBackEndInstance(Model model, HttpServletRequest request);
+    public ResponseEntity<String> deleteBackEndInstance(Model model, HttpServletRequest request) {
+        try {
+            String body = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
```

Comment:
I think you can make clearer here also, what data in Body you use or what body contain.

Reply:
Or document what it contains.

Reply:
renamed